Sources/Public/Animation/LottiePlaybackMode.swift

```diff
@@ -20,6 +27,9 @@ public enum LottiePlaybackMode: Hashable {
   /// Any existing animation will be paused at the current frame.
   case pause
 
+  /// Pauses the animation at a given marker and position
+  case pausedAtMarker(_ marker: String, position: MarkerPosition)
```

Comment:
I'm not totally sure about the `pausedAt` prefix considering we don't use that on the other specific-frame cases. Perhaps instead we call this `startFrameOfMarker(_:)` and `endFrameOfMarker(_:)`?

Reply:
> Thought on restructuring the PlaybackMode enum? This is a bigger change which IMO could clean things up quite a bit (we could leave current APIs in-place marked as deprecated until the next major version bump):

I see the appeal of this sort of change from a semantic perspective. 

I'm especially interested how this API looks at the call-site. For example, I generally prefer:

```swift
.play(.fromProgress(0.5, toProgress: 1)

.play(.frame(currentFrame))
```

over nested forms like:

```swift
.play(.play(.fromProgress(0.5, toProgress: 1))

.play(.pause(.frame(currentFrame)))
```

Reply:
I could imagine splitting `LottiePlaybackMode` into two separate types so you use `.play(...)` for the modes that trigger playback and `.pause(...)` for the modes that don't. I'd be interested in seeing how that affects the syntax for cases like `AnimationPreviewView` where you dynamically switch between playing and paused using an `@State var`.

Reply:
Are you referring to the `CALayer` methods?

@calda I'd imagine we'd add the following in order to unnest the APIs - what do you think?
```
func pause(_ paused: LottiePlaybackMode.Paused)
func play(_ playRange: LottiePlaybackMode.PlayRange)
```

Reply:
Yeah I don't think I mind this change. I acually kind of like the extra nesting here:

Before:

```swift
  private var playbackMode: LottiePlaybackMode {
    if animationPlaying {
      return .fromProgress(playFromProgress, toProgress: playToProgress, loopMode: loopMode)
    } else {
      return .progress(sliderValue)
    }
  }
```

After:

```swift
  private var playbackMode: LottiePlaybackMode {
    if animationPlaying {
      return .play(.fromProgress(playFromProgress, toProgress: playToProgress, loopMode: loopMode))
    } else {
      return .paused(at: .progress(sliderValue))
    }
  }
```

Reply:
We already have `LottieView.playbackMode(_ playbackMode: LottiePlaybackMode)` which takes the root enum. Is this what you might be looking for?